tests/handlers/test_freshmaker.py

```diff
@@ -77,6 +77,65 @@ def test_event_to_building(mock_koji_cs):
     assert build.state == 0
 
 
+@mock.patch('koji.ClientSession')
+def test_event_to_complete(mock_koji_cs):
+    """Test the Freshmaker handler when it receives an event to complete message."""
+    mock_koji_session = mock.Mock()
+    mock_koji_session.getTaskResult.side_effect = [
+        {
+            'koji_builds': ['710916']
+        },
+        {
+            'koji_builds': ['123456']
+        },
+        {
+            'koji_builds': ['234567']
+        }
+    ]
+    mock_koji_cs.return_value = mock_koji_session
+    event = FreshmakerEvent.get_or_create({
+        'id_': '2194',
+        'state': 1,
+        'state_name': 'BUILDING',
+        'url': 'http://freshmaker.domain.com/api/1/events/2194',
+        'event_type_id': 8,
+        'message_id':
+            'ID:messaging.domain.com-42045-1527890187852-9:1045742:0:0:1.RHBA-8018:0600-01'
+    })[0]
+    # Load the message to pass to the handler
+    with open(path.join(message_dir, 'freshmaker', 'event_to_complete.json'), 'r') as f:
+        msg = json.load(f)
+    # Make sure the handler can handle the message
+    assert FreshmakerHandler.can_handle(msg) is True
+    # Instantiate the handler
+    handler = FreshmakerHandler(config)
+    # Run the handler
+    handler.handle(msg)
+    event = FreshmakerEvent.nodes.get_or_none(id_='2194')
+    assert event is not None
+
+    # We have already tested all of the other properties for an event state change, so we will just
+    # ensure that this different type of state change functions properly
+    assert event.state == 2
+    assert event.state_name == 'COMPLETE'
+
+    build = ContainerKojiBuild.nodes.get_or_none(id_=710916)
```

Comment:
Since the message being processed is an "event to complete" message, we can assume that an "event to building" and "build state change" messages were processed previously in the real world, therefore, these container builds should already be in Neo4j and up to date to mimic that. With that said, you'll need to create container builds in your test prior to running the handler, similar to what you are doing with the Freshmaker event. Then after the handler is run, you need to just verify that the container builds haven't changed.